### PR TITLE
add left right support for trigger menu

### PR DESCRIPTION
**Pull request by razvanip**
Closes https://github.com/adobe/react-spectrum/issues/1125

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues/1125).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Review by snowystinger — COMMENTED** [review]
Hey, thanks for starting this. Would you mind adding some storybook examples to show it working as well? You can add them in this file https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/menu/stories/MenuTrigger.stories.tsx

**Comment by razvanip** [comment]
> Hey, thanks for starting this. Would you mind adding some storybook examples to show it working as well? You can add them in this file https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/menu/stories/MenuTrigger.stories.tsx

**Comment by razvanip** [comment]
@snowystinger added some stories, sorry for closing and reopening , wrong button click

File: packages/@react-types/menu/src/index.d.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export interface MenuTriggerProps extends OverlayTriggerProps {
    * Where the Menu opens relative to its trigger.
    * @default 'bottom'
    */
-  direction?: 'bottom' | 'top',
+  direction?: 'bottom' | 'top' | 'left' | 'right',
```
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-types/menu/src/index.d.ts:28]: Start and end should follow how flex's align start and end work. In LTR start is on the left, in RTL start is on the right

**Review by snowystinger — COMMENTED** [review]
I think we're going to need some code changes as well. I just looked at the stories you added (thank you!) and in both cases the menu opens upward on the side. (note, image looks like there wasn't enough room at the bottom, there was, i just didn't want to include a bunch of empty space :) )
Screen Shot 2021-05-07 at 11 36 06 AM

It should open downwards on the side whenever possible, only going up if there isn't enough room. I think this can be accomplished by setting the `align` when we're placed on the sides to 'top'. I'm not sure why it's not working already though, the align is set to 'start' by default, which should produce the same effect.

**Review by snowystinger — APPROVED** [review]
LGTM tested in chrome

**Pull request merged**